library/os_container_infra.py

```diff
@@ -0,0 +1,136 @@
+                patch = dict(op='replace',
+                     value=kwargs['master_count'],
+                     path='/master_count')
+                patch_list.append(patch)
+
+            # Update number of nodes if there is a mismatch
+            node_count_mismatch = kwargs['node_count'] != container_infra_details.node_count
+            if node_count_mismatch:
+                patch = dict(op='replace',
+                     value=kwargs['node_count'],
+                     path='/node_count')
+                patch_list.append(patch)
+
+            if patch_list:
+                # There appears to be a patch needed to apply
+                magnum_client.clusters.update(container_infra_details.uuid, patch_list)
+                display.warning("Applying patch to the cluster: {}".format(patch_list))
+                return False
+            else:
+                # Everything looks okay, return cluster details
+                return container_infra_details
+        else:
+            module.fail_json(
+                msg="Cluster returned unexpected status: {}".format(container_infra_details.status))
+    except NotFound:
+        # Create new cluster here if no cluster with the name was found
+        magnum_client.clusters.create(**kwargs)
+        display.warning("Cluster called {} does not exist, creating new.".format(kwargs['name']))
+        return False
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            cloud_name=dict(required=True, type='str'),
+            magnum_endpoint=dict(required=True, type='str'),
+            cluster_name=dict(required=True, type='str'),
+            cluster_template_name=dict(required=True, type='str'),
+            master_count=dict(required=True, type='int'),
+            node_count=dict(required=True, type='int'),
+            keypair=dict(required=True, type='str'),
+        ),
+        supports_check_mode=False
+    )
+
+    cloud_name = module.params['cloud_name']
+    magnum_endpoint = module.params['magnum_endpoint']
+    os_cloud = get_os_cloud(cloud_name,  module)
+    magnum_client = Client('1', endpoint_override=magnum_endpoint, session=os_cloud.session)
+
+    cluster_template_name = module.params['cluster_template_name']
+    cluster_template_id = magnum_client.cluster_templates.get(cluster_template_name).uuid
+
+    kwargs = dict(name=module.params['cluster_name'],
+                    cluster_template_id=cluster_template_id,
+                    master_count=module.params['master_count'],
+                    node_count=module.params['node_count'],
+                    keypair=module.params['keypair'],
+                )
+
+    tries = 0
+    fail_after = 100
+    changed = False
+    while True:
+        container_infra_details = get_update_or_create_container_infra(magnum_client, module, **kwargs)
+        display.warning(container_infra_details)
+        if container_infra_details:
+            stack_id = container_infra_details.stack_id
```

Comment:
Wouldn't the cluster ID be more useful than the stack_id?

Reply:
we want to go from cluster name->cluster inventory. as a result, i think stack_id is the next best place to look.